nilearn/_utils/bids.py

```diff
@@ -0,0 +1,217 @@
+    if fname in ["fetch_atlas_basc_multiscale_2015"]:
+        index = []
+        for x in name:
+            tmp = x if isinstance(x, str) else int(x)
+            index.append(tmp)
+    elif fname in ["fetch_atlas_schaefer_2018", "fetch_atlas_pauli_2017"]:
+        index = list(range(1, len(name) + 1))
+
+    if len(name) != len(index):
+        if strict:
+            raise ValueError(
+                f"'name' ({len(name)}) and 'index' ({len(index)}) "
+                "have different lengths."
+                "Cannot generate a look upt table."
+            )
+
+        if len(name) < len(index):
+            warnings.warn(
+                "Too many indices for the names."
+                "Padding 'names' with 'unknown'.",
+                stacklevel=3,
+            )
+            name += ["unknown"] * (len(index) - len(name))
+
+        if len(name) > len(index):
+            warnings.warn(
+                "Too many names for the indices.Dropping excess names values.",
+                stacklevel=3,
+            )
+            name = name[0 : len(index)]
+
+    # convert to dataframe and do some cleaning where required
+    lut = pd.DataFrame({"index": index, "name": name})
+
+    if fname in [
+        "fetch_atlas_pauli_2017",
+    ]:
+        lut = pd.concat(
+            [pd.DataFrame([[0, "Background"]], columns=lut.columns), lut],
+            ignore_index=True,
+        )
+
+    return lut
+
+
+def check_look_up_table(lut, atlas, strict=False, verbose=1):
+    """Validate atlas look up table (LUT).
+
+    Make sure it complies with BIDS requirements.
+
+    Throws warning / errors:
+    - lut is not a dataframe with the required columns
+    - if there are mismatches between the number of ROIs
+      in the LUT and th number of unique ROIs in the associated image.
+
+    Parameters
+    ----------
+    lut : :obj:`pandas.DataFrame`
+        Must be a pandas dataframe with at least "name" and "index" columns.
+
+    atlas : Niimg like object or SurfaceImage or numpy array
+
+    strict : bool, default = False
+        Errors are raised instead of warnings if strict == True.
+
+    verbose: int
+        No warning thrown if set to 0.
+
+    Raises
+    ------
+    AssertionError
+        If:
+        - lut is not a dataframe with the required columns
+        - if there are mismatches between the number of ROIs
+          in the LUT and th number of unique ROIs in the associated image.
+
+    ValueError
+        If regions in the image do not exist in the atlas lookup table
+        and `strict=True`.
+
+    Warns
+    -----
+    UserWarning
+        If regions in the image do not exist in the atlas lookup table
+        and `strict=False`.
+
+    """
+    assert isinstance(lut, pd.DataFrame)
+    assert "name" in lut.columns
+    assert "index" in lut.columns
+
+    roi_id = _get_indices_from_image(atlas)
+
+    if len(lut) != len(roi_id):
+        if missing_from_image := set(lut["index"].to_list()) - set(roi_id):
```

Comment:
Just for my clarification, this defines `missing_from_image` variable and is True if the set isn't empty?

Reply:
I see. So in this case, `missing_from_image` would be True if `set(lut["index"].to_list()) - set(roi_id)` isn't empty, right?

Reply:
to be pendantic `missing_from_image` would be a **Truthy** in case `set(lut["index"].to_list()) - set(roi_id)` is not empty

Reply:
Ok cool, I understand better now.

(Ref for anyone else reading this -- Truthy vs True: https://stackoverflow.com/questions/39983695/what-is-truthy-and-falsy-how-is-it-different-from-true-and-false)

Reply:
Thanks for the explanation @Remi-Gau !